twill-core/src/main/java/org/apache/twill/internal/DefaultResourceReport.java

```diff
@@ -149,6 +161,51 @@ public void setServices(List<String> services) {
     return services.get();
   }
 
+  /**
+   * Get the map of the log level arguments of the twill application.
+   *
+   * @return the map of the log level arguments of the twill application
+   */
+  @Override
+  public Map<String, Map<String, LogEntry.Level>> getLogLevels() {
+    return logLevels;
+  }
+
+  /**
+   * Save the log levels for the twill runnable.
+   *
+   * @param runnableName name of the runnable.
+   * @param logLevels map of the log level arguments to be saved.
+   */
+  public void setLogLevels(String runnableName, Map<String, LogEntry.Level> logLevels) {
+    Map<String, LogEntry.Level> newlogLevels = new ConcurrentHashMap<>(logLevels);
```

Comment:
Yes. There will be no concurrent calls since this method is called in the wrap of `ContainerLock`. But immutable map cannot be used since we need to put values to the map if there is update